server/src/main/java/org/opensearch/action/search/StreamSearchActionListener.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.search.SearchPhaseResult;
+import org.opensearch.search.SearchShardTarget;
+
+/**
+ * This class extends SearchActionListener while providing streaming capabilities.
+ *
+ * @param <T> the type of SearchPhaseResult this listener handles
+ */
+abstract class StreamSearchActionListener<T extends SearchPhaseResult> extends SearchActionListener<T> {
+
+    protected StreamSearchActionListener(SearchShardTarget searchShardTarget, int shardIndex) {
+        super(searchShardTarget, shardIndex);
+    }
+
+    /**
+     * Handle intermediate streaming response by preparing it and delegating to innerOnStreamResponse.
+     * This provides the streaming capability for search operations.
+     */
+    public final void onStreamResponse(T response, boolean isLast) {
+        assert response != null;
+        response.setShardIndex(requestIndex);
+        setSearchShardTarget(response);
+        if (isLast) {
+            innerOnCompleteResponse(response);
+            return;
+        }
+        innerOnStreamResponse(response);
+    }
+
+    /**
+     * Handle regular SearchActionListener response by delegating to innerOnCompleteResponse.
+     * This maintains compatibility with SearchActionListener while providing streaming capability.
+     */
+    @Override
+    protected void innerOnResponse(T response) {
```

Comment:
Can we make `innerOnResponse` non abstract in `SearchActionListener`? So we wont have to implement this?